pkg/antctl/raw/upgrade/apistorage/command_test.go

```diff
@@ -0,0 +1,117 @@
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+var (
+	antreaCRDAlfa       = createMockCRD("Alfa", "AlfaList", "alfas", "alfa", "crd.antrea.io", map[string]bool{"v1alpha1": false, "v1beta1": true})
+	antreaCRDBravo      = createMockCRD("Bravo", "BravoList", "bravos", "bravo", "crd.antrea.io", map[string]bool{"v1beta1": true})
+	nonAntreaCRDCharlie = createMockCRD("Charlie", "CharlieList", "charlies", "charlie", "crd.misc.io", map[string]bool{"v1beta1": true})
+	alfa1               = createMockCRDObject("crd.antrea.io/v1beta1", "Alfa", "alfa1")
+	alfa2               = createMockCRDObject("crd.antrea.io/v1beta1", "Alfa", "alfa2")
+	bravo1              = createMockCRDObject("crd.antrea.io/v1beta1", "Bravo", "bravo1")
+)
+
+func createMockCRD(kind, listKind, plural, singular, group string, versions map[string]bool) *apiextv1.CustomResourceDefinition {
+	crd := &apiextv1.CustomResourceDefinition{}
+	var crdVersions []apiextv1.CustomResourceDefinitionVersion
+	var crdStoredVersions []string
+	for version, served := range versions {
+		crdVersions = append(crdVersions, apiextv1.CustomResourceDefinitionVersion{
+			Name:    version,
+			Served:  served,
+			Storage: served,
+		})
+		crdStoredVersions = append(crdStoredVersions, version)
+	}
+
+	crd.SetName(fmt.Sprintf("%s.%s", plural, group))
+	crd.Spec = apiextv1.CustomResourceDefinitionSpec{
+		Group: group,
+		Names: apiextv1.CustomResourceDefinitionNames{
+			Kind:     kind,
+			ListKind: listKind,
+			Plural:   plural,
+			Singular: singular,
+		},
+		Versions: crdVersions,
+	}
+	crd.Status.StoredVersions = crdStoredVersions
+	return crd
+}
+
+func createMockCRDObject(apiVersion, kind, name string) *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(apiVersion)
+	obj.SetNamespace("default")
+	obj.SetName(name)
+	obj.SetKind(kind)
+	return obj
+}
+
+func TestUpgradeCRDs(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		crdsToUpgrade  []string
+		originalCRDs   []client.Object
+		expectedOutput string
+	}{
+		{
+			name:         "Upgrade all CRDs",
+			originalCRDs: []client.Object{antreaCRDAlfa, antreaCRDBravo, nonAntreaCRDCharlie, alfa1, alfa2, bravo1},
+		},
+		{
+			name:          "Upgrade some CRDs",
+			crdsToUpgrade: []string{"bravos.crd.antrea.io"},
+			originalCRDs:  []client.Object{antreaCRDAlfa, antreaCRDBravo, nonAntreaCRDCharlie, alfa1, alfa2, bravo1},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			buf := new(bytes.Buffer)
+			cmd := NewCommand()
+			cmd.SetOut(buf)
+			cmd.SetErr(buf)
+
+			scheme := runtime.NewScheme()
+			apiextinstall.Install(scheme)
+			opts.k8sClient = fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(tt.originalCRDs...).
+				Build()
+
+			if len(tt.crdsToUpgrade) != 0 {
+				args := []string{"crd_list"}
+				args = append(args, tt.crdsToUpgrade...)
+				cmd.SetArgs(args)
+			}
+
+			cmd.Execute()
```

Comment:
I don't get this test, you execute the command but what's the expectation for different cases?
You may check and refer to other commands' tests.

Reply:
Not finished the test yet.